### PR TITLE
feat: singularity overlay create for OCI-SIF

### LINKED ISSUE #2867: add extfs overlay to OCI-SIF via `singularity overlay`
The `singularity overlay create` command should be extended to add an extfs overlay to an OCI-SIF image, in a similar way as is done for a native mode SIF.

The overlay added to the OCI-SIF should be included in an updated image manifest / config, with content-hashes that are consistent for the initial state of the overlay.

### PR DESCRIPTION
**Pull request by dtrudg**
## Description of the Pull Request (PR):

Support `singularity overlay create` for OCI-SIF images.

An extfs overlay file is created, and then appended to the image encapsulated in the OCI-SIF, as an additional layer.

### This fixes or addresses the following GitHub issues:

 - Fixes #2867 

#### Before submitting a PR, make sure you have done the following:

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/main/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/main/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/main/CONTRIBUTING.md)
- Added tests to validate this PR, linted with `make check`  and tested this PR locally with a `make test`, and `make testall` if possible (see CONTRIBUTING.md).
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/main/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/main/CONTRIBUTORS.md)

### CODE REVIEW

File: internal/pkg/ocisif/overlay.go [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+)
+
+var Ext3LayerMediaType types.MediaType = "application/vnd.sylabs.image.layer.v1.ext3"
+
+// HasOverlay returns whether the OCI-SIF at imgPath has an ext3 writable final
+// layer - an 'overlay'.
+func HasOverlay(imagePath string) (bool, error) {
+	fi, err := sif.LoadContainerFromPath(imagePath,
+		sif.OptLoadWithFlag(os.O_RDONLY),
+		sif.OptLoadWithCloseOnUnload(false),
```
Comment (tri-adam) [inline internal/pkg/ocisif/overlay.go:29]: Think you probably _do_ want to close the backing file on `UnloadContainer`, so the file handle doesn't leak?

```suggestion
```

File: internal/pkg/ocisif/overlay.go [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+	img, err := ii.Image(imageDigest)
+	if err != nil {
+		return false, fmt.Errorf("while initializing image: %w", err)
+	}
+
+	layers, err := img.Layers()
+	if err != nil {
+		return false, fmt.Errorf("while getting image layers: %w", err)
+	}
+	mt, err := layers[len(layers)-1].MediaType()
```
Comment (tri-adam) [inline internal/pkg/ocisif/overlay.go:59]: Should probably sanity check that `len(layers) > 0` before doing this just to be safe?

**Review by tri-adam — COMMENTED** [review]
LGTM, just a couple small suggestions. Thanks!

File: internal/pkg/ocisif/overlay.go (hunk 1 above)
Comment (dtrudg) [reply internal/pkg/ocisif/overlay.go:29]: Yikes.. thanks... copied and pasted that from somewhere 😬

**Review by tri-adam — APPROVED** [review]

**Pull request merged**